firefox_ui_tests/remote/security/test_safe_browsing_notification.py

```diff
@@ -0,0 +1,150 @@
+
+                # Go back to the notification bar
+                self.marionette.navigate(unsafe_page)
+                self.check_ignore_warning_button(unsafe_page)
+
+                self.marionette.set_context('chrome')
+                self.check_x_button()
+                self.marionette.set_context('content')
+
+    def check_ignore_warning_button(self, unsafe_page):
+        ignore_warning_button = self.marionette.find_element(By.ID, "ignoreWarningButton")
+
+        # This isn't clickable by the time we get here and needs a
+        # pretty significant delay.
+        time.sleep(1)
+
+        ignore_warning_button.click()
+
+        def find_main_feature_el(mn):
+            try:
+                mn.find_element("id", "main-feature")
+                return True
+            except:
+                return False
+
+        self.wait_for_condition(find_main_feature_el)
+        self.assertRaises(NoSuchElementException, self.marionette.find_element,
+                          'id', 'ignoreWarningButton')
+        self.assertEquals(self.marionette.get_url(), unsafe_page)
+        self.perms.remove('www.itisatrap.org', 'safe-browsing')
+
+    def find_button(self, element, label):
+        buttons = element.find_elements('tag name', 'button')
+        found_button = None
+        for button in buttons:
+            if button.get_attribute('label') == label:
+                found_button = button
+        self.assertTrue(found_button is not None, 'Could not find button')
+        return found_button
+
+    def check_no_phishing_button(self, label, report_page):
+        button_label = self.browser.get_property(label)
+        with self.marionette.using_context('chrome'):
+            tab_count = len(self.tabbar.tabs)
+            blocked_badware_page = (self.marionette.find_element(By.ID, 'content')
+                                    .find_element('anon attribute',
+                                    {'value': 'blocked-badware-page'}))
+            self.find_button(blocked_badware_page, button_label).click()
+
+            self.wait_for_condition(lambda _: len(self.tabbar.tabs) == tab_count + 1)
+            self.wait_for_condition(lambda _: report_page in self.browser.navbar.locationbar.value)
+            self.tabbar.close_tab()
+
+    def check_get_me_out_of_here_button(self):
+        label = self.browser.get_property(
+            'safebrowsing.getMeOutOfHereButton.label')
+        with self.marionette.using_context('chrome'):
+            blocked_badware_page = (self.marionette.find_element(By.ID, 'content')
+                                    .find_element('anon attribute',
+                                    {'value': 'blocked-badware-page'}))
+            self.find_button(blocked_badware_page, label).click()
+
+        expected_homepage = self.prefs.get_pref('browser.startup.homepage',
+                                                interface='nsIPrefLocalizedString')
```

Comment:
This needs a property most likely on the BrowserWindow class to get the default homepage.